src/lib/heliaFetch.ts

```diff
@@ -1,17 +1,108 @@
+import { createVerifiedFetch, type ContentTypeParser } from '@helia/verified-fetch'
+import { fileTypeFromBuffer } from '@sgtpooki/file-type'
 import type { Helia } from '@helia/interface'
-import { unixfs } from '@helia/unixfs'
-import { CID } from 'multiformats/cid'
-// import { ipns, ipnsValidator, ipnsSelector } from '@helia/ipns'
-
-import { getDirectoryResponse } from './heliaFetch/getDirectoryResponse.ts'
-import { getFileResponse } from './heliaFetch/getFileResponse.ts'
-import { GetDNSLinkOrIPNS } from '../kubo-rpc-ipns.ts'
 
 export interface HeliaFetchOptions {
   path: string
   helia: Helia
   signal?: AbortSignal
   headers?: Headers
+  origin?: string | null
+  protocol?: string | null
+}
+
+// default from verified-fetch is application/octect-stream, which forces a download. This is not what we want for MANY file types.
+const defaultMimeType = 'text/html; charset=utf-8'
+const contentTypeParser: ContentTypeParser = async (bytes, fileName) => {
+  // eslint-disable-next-line no-console
+  console.log('bytes received in contentTypeParser for ', fileName, ' : ', bytes.slice(0, 10), '...')
+
+  const detectedType = (await fileTypeFromBuffer(bytes))?.mime
+  if (detectedType != null) {
+    return detectedType
+  }
+  if (fileName == null) {
+    // no other way to determine file-type.
+    return defaultMimeType
+  }
+
+  // no need to include file-types listed at https://github.com/SgtPooki/file-type#supported-file-types
+  switch (fileName.split('.').pop()) {
+    case 'css':
+      return 'text/css'
+    case 'html':
+      return 'text/html; charset=utf-8'
```

Comment:
Q: why are we assuming every HTML uses UTF-8?

Reply:
bad habit. removed.